subsys/testsuite/ztest/include/ztest_test.h

```diff
@@ -27,7 +29,65 @@ struct unit_test {
 	uint32_t thread_options;
 };
 
-void z_ztest_run_test_suite(const char *name, struct unit_test *suite);
+/**
+ * Metadata about a ztest suite
+ */
+struct ztest_suite_metadata {
+	uint32_t run_count;
+	uint32_t skip_count;
+	uint32_t fail_count;
+};
+
+/**
+ * A single node of test suite. Each node should be added to unit_test_head to
+ * form a linked list that can be run via ztest_run_registered_test_suites().
+ */
+struct ztest_suite_node {
+	/** The name of the test suite. */
+	const char *name;
+	/** Pointer to the test suite. */
+	struct unit_test *suite;
+	/**
+	 * An optional pragma function to determine if the test should run. If NULL, then the test
+	 * will only run once on the first attempt.
+	 */
+	bool (*pragma)(const void *state);
+	/** Metadata */
+	struct ztest_suite_metadata metadata;
+};
+
+extern const struct ztest_suite_node __ztest_suite_start[];
+extern const struct ztest_suite_node __ztest_suite_end[];
```

Comment:
Must remove `const` since `metadaa`, which resides inside `ztest_suite_node` is mutable.